app/components/osc-webhook-triggers/osc-webhook-triggers.html

```diff
@@ -1,38 +1,55 @@
     <div class="add-webhook-row">
-      <ui-select 
-        ng-model="trigger.data.type"
-        ng-disabled="$ctrl.isDeprecated(trigger)"
-        on-select="$ctrl.triggerTypeChange(trigger)"
-        search-enabled="false"
-        title="Select a webhook type"
-        class="select-webhook-type"
-        flex>
-        <ui-select-match placeholder="Webhook type">
-          {{ $select.selected.label }}
-        </ui-select-match>
-        <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
-          {{ option.label }}
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-webhook-type">
+        <ui-select 
+          ng-model="trigger.data.type"
+          name="triggerType{{$index}}"
+          ng-disabled="$ctrl.isDeprecated(trigger)"
+          on-select="$ctrl.triggerTypeChange(trigger)"
+          search-enabled="false"
+          title="Select a webhook type"
+          ng-class="{'has-error': $ctrl.secretsForm['triggerType{{$index}}'].$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name }"
+          focus-on="triggerTypeFocus{{$index}}"
+          ng-required="!trigger.data.type || trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
+          flex>
+          <ui-select-match placeholder="Webhook type">
+            {{ $select.selected.label }}
+          </ui-select-match>
+          <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
+            {{ option.label }}
+          </ui-select-choices>
+        </ui-select>
+        <div class="has-error" ng-show="$ctrl.secretsForm['triggerType{{$index}}'].$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name">
+          <span class="help-block">
+            Secret type is required.
+          </span>
+        </div>
+      </div>
 
-      <ui-select
-        ng-if="!$ctrl.isDeprecated(trigger)"
-        ng-model="trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
-        ng-disabled="!trigger.data.type"
-        ng-required="trigger.data.type"
-        on-select="$ctrl.triggerSecretChange(trigger)"
-        title="Select a webhook secret reference"
-        class="select-secret-ref"
-        flex>
-        <ui-select-match placeholder="Webhook secret reference">
-          {{ $select.selected.metadata.name }}
-        </ui-select-match>
-        <ui-select-choices repeat="webhookSecret.metadata.name as webhookSecret in ($ctrl.webhookSecrets | filter : $select.search) track by (webhookSecret | uid)">
-          <div ng-bind-html="webhookSecret.metadata.name | highlight : $select.search"></div>
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-secret-ref">
+        <ui-select
+          ng-if="!$ctrl.isDeprecated(trigger)"
+          ng-model="trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
```

Comment:
This could be problematic because you can set a value before trigger.data.type is set, so we'd be setting `trigger.data[''].secretReference.name`. I'm not sure what problems that might cause.

I think it's simpler just to disable the select until the trigger type is selected, which avoids a lot edge cases and unnecessary validation on the type field.

https://github.com/jhadvig/origin-web-console/pull/1/commits/0424279e9228b060b6e49469f7cd61cac9afe2e8#diff-08a15e28e59fecd178b2d160ea9a2362R31